kernel/common/enumerable.rb

```diff
@@ -754,4 +765,13 @@ def include?(obj)
 
   alias_method :member?, :include?
 
+  def to_enum_with_size(method, size, *args)
```

Comment:
I'm not sure if this method belongs in Enumerable itself, we might want to use a mirror for it instead (not sure if they work with modules though). Also, I think it's better if we clean up the mess of `Enumerator#initialize` and `Enumerator#initialize_enumerator` instead of using allocate and the latter. This however is up to @brixen since I'm not 100% sure of it.